tests/src/Interop/PInvoke/ArrayWithOffset/ArrayWithOffsetTest.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text;
+using System.Runtime.InteropServices;
+using TestLibrary;
+
+unsafe class ArrayWithOffsetTest
+{
+    private const string NativeLibrary = "ArrayWithOffsetNative";
+
+    [DllImport(NativeLibrary)]
+    private static extern bool Marshal_InOut(int* expected, [In, Out] ArrayWithOffset actual, int numElements, int* newValue);
+
+    [DllImport(NativeLibrary, EntryPoint = "Marshal_Invalid")]
+    private static extern bool Marshal_In(ArrayWithOffset invalidArray);
```

Comment:
I don't really understand what this is trying to do here. Do you have any argument as to what we are validating/confirming?

Reply:
We're validating that we throw a `MarshalDirectiveException` when we don't have both `[In, Out]` and are just using the default (which I believe shows up as `[In]`.